File: FINAL/2 - Vectors/06 - P71310 - Producte escalar/main.py
from yogi import scan, read

n = scan(int)
while n is not None:
    u = [read(float) for i in range(0, n)]
    v = [read(float) for i in range(0, n)]
    print("{0:.4f}".format(producte_escalar(u, v)))
    n = scan(int)

File: FINAL/2 - Vectors/06 - P71310 - Producte escalar/main.c
#include <stdio.h>

double producte_escalar(int n, double u[n], double v[n]);

int main()
{
    int i, n;
    while (scanf("%d", &n) > 0) {
        double u[n];
        double v[n];
        for (i = 0; i < n; ++i)
            scanf("%lf", &u[i]);
        for (i = 0; i < n; ++i)
            scanf("%lf", &v[i]);
        printf("%.4lf\n", producte_escalar(n, u, v));
    }
    return 0;
}
